### PR TITLE
[WIP] 1.16.200/100 support

### PR DESCRIPTION
**Pull request by CiviledCode**

### CODE REVIEW

**Comment by CiviledCode** [comment]
BlockID issues are fixed, the only issues left are the item runtime Id issues

**Comment by JustTalDevelops** [comment]
passing! 🎉

**Comment by JustTalDevelops** [comment]
At the moment, we have a loader for the new item entries and a map to go between the old ID + meta to the new runtime ID, but we're not sure on how we're going to be able to change the runtime ID back to the old ID and meta and the other way around. This is the only issue we're aware of right now, aside from the commented out blocks that need to be reimplemented.

**Comment by JustTalDevelops** [comment]
At the moment, not all blocks are being translated correctly (I'll have to look into how I can fix that later), but after a small update to Gophertunnel (making a PR in a litttle), the inventory seems to work just fine. All that is left right now is reimplementing the blocks that were commented out, fixing the runtime ID translation for some items, and a general code cleanup.

**Comment by CiviledCode** [comment]
As of right now block registration for multi-state blocks is required and creative items for the create inventory. Tal is currently working out the creative inventory, while I'm finishing the remaining blocks. If anybody is looking for a good starting project, feel free to help contribute to registering the block states

**Comment by cunningr** [comment]
Hi.  I was just trying this fork on Rasp. Pi with client 1.16.201 (Android) but I couldn't get it to connect initially.  After restarting the client did connect but the server immediately crashed;

pi@raspberrypi:~/dragonfly $ go run main.go
INFO[0000] Starting server...
DEBU[0000] Loading world...
DEBU[0001] Loaded world 'World'.
INFO[0001] Server running on [::]:19132.
SnaggyAxis34484 has joined the game
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12394]

goroutine 54 [running]:
runtime/internal/atomic.goLoad64(0x2ce2154, 0x0, 0x0)
	/usr/local/go/src/runtime/internal/atomic/atomic_arm.go:131 +0x1c
go.uber.org/atomic.(*Int64).Load(...)
	/home/pi/go/pkg/mod/go.uber.org/atomic@v1.7.0/int64.go:45
github.com/df-mc/dragonfly/dragonfly/player.(*Player).OnFireDuration(...)
	/home/pi/dragonfly/dragonfly/player/player.go:817
github.com/df-mc/dragonfly/dragonfly/player.(*Player).State(0x2ce2000, 0x6703d8, 0x2ce2000, 0x2caa420)
	/home/pi/dragonfly/dragonfly/player/player.go:1742 +0x2c4
github.com/df-mc/dragonfly/dragonfly/world.showEntity(0x6703d8, 0x2ce2000, 0x6706f0, 0x38000d0)
	/home/pi/dragonfly/dragonfly/world/world.go:1453 +0x48
github.com/df-mc/dragonfly/dragonfly/world.(*World).addViewer(0x2ce2340, 0x292b360, 0x6706f0, 0x38000d0)
	/home/pi/dragonfly/dragonfly/world/world.go:1344 +0xbc
github.com/df-mc/dragonfly/dragonfly/world.(*Loader).Load(0x2d7e190, 0x4, 0x2, 0x0)
	/home/pi/dragonfly/dragonfly/world/loader.go:98 +0x110
github.com/df-mc/dragonfly/dragonfly/session.(*Session).sendChunks(0x38000d0, 0x2d313c0)
	/home/pi/dragonfly/dragonfly/session/session.go:245 +0x154
created by github.com/df-mc/dragonfly/dragonfly/session.(*Session).handlePackets
	/home/pi/dragonfly/dragonfly/session/session.go:212 +0x78
exit status 2

I see it is WIP and although I don't have the skills to debug/fix I am happy to test potential fixes.

File: go.mod [hunk 1]
```diff
@@ -3,24 +3,25 @@ module github.com/df-mc/dragonfly
 go 1.13
 
 require (
-	github.com/brentp/intintmap v0.0.0-20190211203843-30dc0ade9af9
+	github.com/aquilax/go-perlin v1.0.0
```
Comment (T14Raptor) [inline go.mod:6]: This is unused.

File: dragonfly/world/chunk/chunk.go [hunk 2]
```diff
@@ -5,6 +5,8 @@ import (
 	"sync"
 )
 
+var AirRuntimeID uint32
```
Comment (T14Raptor) [inline dragonfly/world/chunk/chunk.go:8]: I'd rather have this in an internal package.

File: go.mod (hunk 1 above)
Comment (JustTalDevelops) [reply go.mod]: removed

File: dragonfly/world/chunk/chunk.go (hunk 2 above)
Comment (JustTalDevelops) [reply dragonfly/world/chunk/chunk.go]: Moved to world internal

**Comment by CiviledCode** [comment]
All bugs are finished, just waiting for a review

File: dragonfly/world/axis.go [hunk 3]
```diff
@@ -39,3 +39,7 @@ func (a Axis) FromString(s string) (interface{}, error) {
 	}
 	return nil, fmt.Errorf("unexpected axis '%v', expecting one of 'x', 'y' or 'z'", s)
 }
+
+func AllAxis() []Axis {
```
Comment (Sandertv) [inline dragonfly/world/axis.go:43]: Should probably name this `AllAxes` (and don't forget to document this).

**Review by Sandertv — CHANGES_REQUESTED** [review]
Got one comment.

File: dragonfly/world/axis.go (hunk 3 above)
Comment (JustTalDevelops) [reply dragonfly/world/axis.go]: Renamed 👍

**Review by Sandertv — APPROVED** [review]

**Comment by JustTalDevelops** [comment]
POG

**Comment by provsalt** [comment]
Tal is goated

**Pull request merged**